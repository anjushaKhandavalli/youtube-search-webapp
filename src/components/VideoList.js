import React from 'react';
import VideoItem from './VideoItem';
import InfiniteScroll from "react-infinite-scroll-component";
import { VideoListContainer } from './VideoItem.style';


const VideoList = ({videos,fetchMoreVideos}) => {
  const renderVideos =  videos.map((video) => {
    return <VideoItem data-testid={video.id.videoId} key={video.id.videoId} video={video}/>
});

return (
    <VideoListContainer data-testid="video-list">
      <InfiniteScroll
        dataLength={videos.length}
        next={fetchMoreVideos}
        hasMore={true}
        loader={videos.length === 0 ? null : <h4>Loading...</h4>}
      >
        {renderVideos}
      </InfiniteScroll>
    </VideoListContainer>
  )
}

export default VideoList;