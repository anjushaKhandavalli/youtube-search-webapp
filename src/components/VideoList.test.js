import React from 'react';
import { render } from "@testing-library/react";
import VideoList from './VideoList';


describe("videoList", () => {
  const mockFetchMoreVideos = jest.fn();
  const mockVideos = [
    {
      id: {
        kind: "youtube#video",
        videoId: "cat_1"
      },
      snippet: {
        title: "cat_1 video title",
        description: "description about the cat_1",
        thumbnails: {
          medium : {
            url: "https://i.ytimg.com/vi/AETFvQonfV8/default.jpg",
          }
        }
      }
    },
    {
      id: {
        kind: "youtube#video2",
        videoId: "cat_2"
      },
      snippet: {
        title: "cat_2 video title",
        description: "description about the cat_2",
        thumbnails: {
          medium : {
            url: "https://i.ytimg.com/vi/AETFvQoabc/default.jpg",
          }
        }
      }
    }
  ]
  it("should render the videolist", () => {
    const {container, getByTestId } = render(<VideoList videos={mockVideos} fetchMoreVideos={mockFetchMoreVideos}/>)

    expect(container).toMatchSnapshot();
    expect(getByTestId('video-list')).toBeDefined();
  })

  it("should not render loading text when there are no videos", () => {
    const { queryByText } = render(<VideoList videos={[]} fetchMoreVideos={mockFetchMoreVideos}/>)

    expect(queryByText("Loading...")).not.toBeInTheDocument;

  })
})