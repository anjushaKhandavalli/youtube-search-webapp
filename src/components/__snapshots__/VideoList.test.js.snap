// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`videoList should render the videolist 1`] = `
<div>
  <div
    class="sc-jSgupP cQxPOd"
    data-testid="video-list"
  >
    <div
      class="infinite-scroll-component__outerdiv"
    >
      <div
        class="infinite-scroll-component "
        style="height: auto; overflow: auto;"
      >
        <div
          class="sc-bdfBwQ jwCXbJ"
          data-testid="video-item"
        >
          <img
            alt="description about the cat_1"
            class="sc-gsTCUz dLvDma"
            data-testid="thumbnail"
            src="https://i.ytimg.com/vi/AETFvQonfV8/default.jpg"
          />
          <div
            class="sc-dlfnbm jxTFsF"
          >
            <h2
              class="sc-hKgILt ksesVL"
              data-testid="title"
            >
              cat_1 video title
            </h2>
            <h4
              class="sc-eCssSg gqHjlA"
              data-testid="description"
            >
              description about the cat_1
            </h4>
          </div>
        </div>
        <div
          class="sc-bdfBwQ jwCXbJ"
          data-testid="video-item"
        >
          <img
            alt="description about the cat_2"
            class="sc-gsTCUz dLvDma"
            data-testid="thumbnail"
            src="https://i.ytimg.com/vi/AETFvQoabc/default.jpg"
          />
          <div
            class="sc-dlfnbm jxTFsF"
          >
            <h2
              class="sc-hKgILt ksesVL"
              data-testid="title"
            >
              cat_2 video title
            </h2>
            <h4
              class="sc-eCssSg gqHjlA"
              data-testid="description"
            >
              description about the cat_2
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
